### PR TITLE
Makman2/linterv4-adapt

### PR DESCRIPTION
**Pull request by Makman2**
Adaption of the Linter v2 API

I've divided refactors into several packs for easier review^^

### CODE REVIEW

**Comment by sils1297** [comment]
blocked by https://github.com/coala-analyzer/coala/pull/1990

File: bears/c_languages/CPPLintBear.py [hunk 1]
```diff
@@ -1,32 +1,29 @@
...
+class CPPLintBear:
+    """
+    Checks the code with ``cpplint`` on each file separately.
+    """
 
+    @staticmethod
+    def create_arguments(filename, file, config_file,
+                         max_line_length: int=80,
+                         cpplint_ignore: typed_list(str)=[],
+                         cpplint_include: typed_list(str)=[]):
```
Comment (sils1297) [inline bears/c_languages/CPPLintBear.py]: filed https://github.com/coala-analyzer/coala-bears/issues/345 about mutable default params

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/313/commits/55fde26e4aec8aa4a84f25df72689b12c9efaf3f https://github.com/coala-analyzer/coala-bears/pull/313/commits/f59358d407e3395e4c659aff768f6ff008e0ace0 https://github.com/coala-analyzer/coala-bears/pull/313/commits/e34bea0d8f448594cc4b710cd33fa7a1820760c0

File: bears/c_languages/GNUIndentBear.py [hunk 2]
```diff
@@ -1,38 +1,35 @@
...
-        self.arguments = "--no-tabs" if use_spaces else "--use-tabs"
-        self.arguments += (" --line-length {0} --indent-level {1} "
-                           "--tab-size {1} {2}".format(max_line_length,
-                                                       tab_width,
-                                                       indent_cli_options))
-        return self.lint(filename, file)
+        return ("--no-tabs" if use_spaces else "--use-tabs",
+                "--line-length", str(max_line_length),
+                "--indent-level", str(tab_width),
+                "--tab-size", str(max_line_length), str(indent_cli_options))
```
Comment (sils1297) [inline bears/c_languages/GNUIndentBear.py]: the tab width was used two times in the old version! Your new parameters set the line length as the tab size

**Comment by sils1297** [comment]
unack https://github.com/coala-analyzer/coala-bears/pull/313/commits/e1e65d032123ede271db7aabbcf387176692606a

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/313/commits/be6d9a6837493da815f6df01156f97e52d446a90 https://github.com/coala-analyzer/coala-bears/pull/313/commits/b93f3f0838159aa80d85833fba92ccec922a373e https://github.com/coala-analyzer/coala-bears/pull/313/commits/2e6a7827275888af61e81a18efc25b1a4d09f674 https://github.com/coala-analyzer/coala-bears/pull/313/commits/5cb84927c466ab7a37ecf066b7f5210c8c013416 https://github.com/coala-analyzer/coala-bears/pull/313/commits/09bc30b2fc1e5e0c4536a92aca744c0b2dc69369

Btw. @Makman2 you are allowed to reack commits when just changing `Linter` to `linter` which is obviously needed.

**Comment by sils1297** [comment]
ack 09bc30b

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/313/commits/9c3dd65b7173b835686e447f05aef733eaa92e80 https://github.com/coala-analyzer/coala-bears/pull/313/commits/4365ab02047a707e2405b382f0ca0e3c9aa5517d https://github.com/coala-analyzer/coala-bears/pull/313/commits/5adedfd3409b753d72fd921e02f9645bc594df3f

**Comment by sils1297** [comment]
@Makman2 taking over the PR now

File: bears/xml/XMLBear.py [hunk 3]
```diff
@@ -1,53 +1,58 @@
...
+                self.process_output_regex(
+                    output[1], filename, file,
+                    output_regex=self.output_regex),
+                self.process_output_corrected(
+                    output[0], filename, file,
+                    diff_message="XML can be formatted better."))
+        else:  # Return issues from stderr if stdout is empty
+            return self.process_output_regex(
+                output[0], filename, file,
+                output_regex=self.output_regex)
```
Comment (sils1297) [inline bears/xml/XMLBear.py]: something's wrong here, not taking it into my PRs, we'll look at it later. (Tests fail)

File: bears/xml/XMLBear.py [hunk 4]
```diff
@@ -1,54 +1,58 @@
...
-from coalib.bears.LocalBear import LocalBear
-from coalib.misc.Shell import escape_path_argument
+from coalib.bearlib.abstractions.Linter import linter
 from coalib.settings.Setting import path, url
 
 
 def path_or_url(xml_dtd):
-    '''
+    """
     Coverts the setting value to url or path.
```
Comment (Adrianzatreanu) [inline bears/xml/XMLBear.py]: newcomer issue? typo here? `Coverts`

Comment (Makman2) [inline bears/xml/XMLBear.py]: yes :)

**Pull request closed without merging**